core/src/main/java/org/apache/carbondata/core/util/comparator/Comparator.java

```diff
@@ -25,24 +25,34 @@
 
 public final class Comparator {
 
+  //Comparators are made static so that only one instance is generated
+  private static SerializableComparator Boolean  = new BooleanSerializableComparator();
+  private static SerializableComparator Int = new IntSerializableComparator();
+  private static SerializableComparator Short = new ShortSerializableComparator();
+  private static SerializableComparator Double = new DoubleSerializableComparator();
+  private static SerializableComparator Float = new FloatSerializableComparator();
+  private static SerializableComparator Long = new LongSerializableComparator();
+  private static SerializableComparator Decimal  = new BigDecimalSerializableComparator();
+  private static SerializableComparator Byte = new ByteArraySerializableComparator();
+
   public static SerializableComparator getComparator(DataType dataType) {
```

Comment:
Please merge both the methods

Reply:
getComparatorByDataTypeForMeasure method is been used extensively in the codebase making the refactoring difficult and thus cannot be merged, I am in turn using the other function in the first function.